acceptance/tests/pxp-module-puppet/run_puppet_twice.rb

```diff
@@ -0,0 +1,95 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'Run Puppet while a Puppet Agent run is in-progress, wait for completion' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  env_name = test_file_name = File.basename(__FILE__, '.*')
+  environment_name = mk_tmp_environment(env_name)
+
+  step 'On master, create a new environment that will result in a slow run' do
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  case $::osfamily {
+    'windows': { exec { 'sleep':
+                        command => 'true',
+                        unless  => 'sleep 10', #PUP-5806
```

Comment:
PUP-5806 is now closed, it advises that the Windows command should be:
    command => 'cmd.exe /c sleep 5 || true'
(That might have the undesirable behaviour of not having the sleep process appear in cygwin's ps though)